src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -2450,6 +2472,74 @@ FindOrCreateWorkerSession(WorkerPool *workerPool, MultiConnection *connection)
 }
 
 
+/*
+ * RemoteSocketClosedForNewSession is a helper function for detecting whether
+ * the remote socket corresponding to the input session is closed. This is
+ * mostly common there is a cached connection and remote server restarted
+ * (due to failover or restart etc.).
+ *
+ * The function is not a generic function that can be called at the start of
+ * the execution.
```

Comment:
Why is it not a generic function?

Reply:
There is nothing specific I'm concerned. But this function doesn't check for interrupts or does not wait at all. So, the callers should be careful not to expect those things

Reply:
probably good to add those examples to the comment